src/client/debugger/jupyter/debuggingManager.ts

```diff
@@ -232,6 +212,51 @@ export class DebuggingManager implements IExtensionSingleActivationService, IDeb
         this.runByLineInProgress.set(runningByLine).ignoreErrors();
     }
 
+    private async tryToStartDebugging(mode: KernelDebugMode, editor?: NotebookEditor, cell?: NotebookCell) {
+        if (!editor) {
+            void this.appShell.showErrorMessage(DataScience.noNotebookToDebug());
+            return;
+        }
+
+        if (this.notebookInProgress.has(editor.document) || this.isDebugging(editor.document)) {
+            return;
+        }
+
+        try {
+            this.notebookInProgress.add(editor.document);
+            if (
+                await this.checkForIpykernel6(
+                    editor.document,
+                    mode === KernelDebugMode.RunByLine ? DataScience.startingRunByLine() : undefined
+                )
+            ) {
+                switch (mode) {
+                    case KernelDebugMode.Everything:
+                        this.updateToolbar(true);
+                        await this.startDebugging(editor.document);
+                        break;
+                    case KernelDebugMode.Cell:
+                        if (cell) {
+                            this.updateToolbar(true);
+                            await this.startDebuggingCell(editor.document, KernelDebugMode.Cell, cell);
```

Comment:
I'm sorry, but I'm not convinced we need `awaits` here
The old code didn't have awaits, feels like there's something we're not understanding here 
I.e. why do we need an await or what happens if there is no await?

Would like to have some comments why we must await, e.g. tomorrow what would stop working if we removed it.

Reply:
it helps with keeping the context keys aligned with the status of the debugger. My most recent commit puts them after the awaits.